amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/DialectsParser.scala

```diff
@@ -45,21 +42,39 @@ class DialectsParser(root: Root)(implicit override val ctx: DialectContext)
 
   // Need to do this before parsing so every ID set during parsing is relative to this ID
   private def id(): String = {
-    root.location match {
-      case AMFParser.DEFAULT_DOCUMENT_URL =>
-        val nameAndVersionId = for {
-          dialectEntry <- map.key("dialect")
-          versionEntry <- map.key("version")
-        } yield {
-          val dialect = dialectEntry.value.toString.toLowerCase.noSpaces
-          val version = versionEntry.value.toString.toLowerCase.noSpaces
-          s"${AMFParser.DEFAULT_DOCUMENT_URL}/$dialect/$version"
+    map.key("$id") match {
```

Comment:
Just as an afterthough, I'm tempted to move this out of the DialectsParser -> out of sight, out of mind